src/lexer.coffee

```diff
@@ -141,34 +141,38 @@ exports.Lexer = class Lexer
       else
         'IDENTIFIER'
 
-    if tag is 'IDENTIFIER' and (id in JS_KEYWORDS or id in COFFEE_KEYWORDS) and
-       not (@exportSpecifierList and id in COFFEE_KEYWORDS)
-      tag = id.toUpperCase()
-      if tag is 'WHEN' and @tag() in LINE_BREAK
-        tag = 'LEADING_WHEN'
-      else if tag is 'FOR'
-        @seenFor = yes
-      else if tag is 'UNLESS'
-        tag = 'IF'
-      else if tag is 'IMPORT'
-        @seenImport = yes
-      else if tag is 'EXPORT'
-        @seenExport = yes
-      else if tag in UNARY
-        tag = 'UNARY'
-      else if tag in RELATION
-        if tag isnt 'INSTANCEOF' and @seenFor
-          tag = 'FOR' + tag
-          @seenFor = no
-        else
-          tag = 'RELATION'
-          if @value() is '!'
-            poppedToken = @tokens.pop()
-            id = '!' + id
-
-    if tag is 'IDENTIFIER' and id in RESERVED
-      @error "reserved word '#{id}'", length: id.length
-
+    if tag is 'IDENTIFIER'
+      if @seenFor and id is 'from'
```

Comment:
Can this be refactored so that we don't branch this `if` block again? In other words, don't indent the following 15ish lines.